torch/distributed/_functional_collectives_impl.py

```diff
@@ -87,6 +87,13 @@ def _wait_tensor(tensor: torch.Tensor) -> torch.Tensor:
         wait_reg.wait()
     return tensor
 
+def _tensor_needs_wait(tensor: torch.Tensor) -> Optional[bool]:
+    data_ptr = tensor.data_ptr()
```

Comment:
do you need: `global data_ptr_to_work` here?

Reply:
no, you only need to use `global` if you assign to it.

Reply:
Interesting, because we use global in L83, too. That's why I am asking.